clang/lib/Sema/HLSLBuiltinTypeDeclBuilder.cpp

```diff
@@ -453,11 +481,11 @@ BuiltinTypeMethodBuilder &BuiltinTypeMethodBuilder::dereference(T Ptr) {
   return *this;
 }
 
-BuiltinTypeDeclBuilder &BuiltinTypeMethodBuilder::finalizeMethod() {
+BuiltinTypeDeclBuilder &BuiltinTypeMethodBuilder::finalize() {
   assert(!DeclBuilder.Record->isCompleteDefinition() &&
          "record is already complete");
-  assert(Method != nullptr &&
-         "method decl not created; are you missing a call to build the body?");
+
+  ensureCompleteDecl();
```

Comment:
Should we add a parameter here to assert if the decl hasn't been created to maintain this assertion behavior?

Reply:
I don't think it is necessary since it is exactly what `ensureCompleteDecl` is doing - making sure Method is created.

Reply:
Previously this assert would fail if the method or constructor had no body statements, and therefore no Method decl was created. Updating it to call `ensureCompleteDecl` created the decl and allows the body to be empty, which is what the current default constructor on resource classes has.